code/src/Components/AddTask/CreateNewTask.js

```diff
@@ -0,0 +1,169 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import styled from "styled-components";
+
+import { todos } from "../../Reducers/todos";
+import { SelectCategory } from "./SelectCategory";
+import { SelectDeadline } from "./SelectDeadline";
+
+export const CreateNewTask = () => {
+  //variables
+  const dispatch = useDispatch();
+  const filters = useSelector((store) => store.todos.filters[0]);
+  const sort = useSelector((store) => store.todos.sort[0]);
+  const cat = useSelector((store) => store.todos.category);
+  const currentTime = new Date(Date.now()).toLocaleString();
+
+  //states
+  const [newTask, setNewTask] = useState("");
+  const [due, setDue] = useState("3030-30-30");
+  const [category, setCategory] = useState("");
+  const [checked, setChecked] = useState("");
+  const [hasDeadline, setHasDeadline] = useState(false);
+
+  //local functions
+  const handleClick = () => {
+    if (newTask === "") {
+      alert("You must add text!");
+    } else {
+      dispatch(
+        todos.actions.createNewTodo({
+          task: newTask,
+          time: currentTime,
+          deadline: hasDeadline ? due : "3030-30-30", //setting a very late date on the ones with no dadline to sort properly.
+          category: category,
+          isTimed: hasDeadline,
+          overdue: new Date(due) - new Date(Date.now()) <= 0 ? true : false,
```

Comment:
I decide to eat Pierogi tomorrow and they are already overdue? And to eat Bigos 10 days from today I need to make it  26 weeks from today :( Check your calculations or your reviewer will starve to death ☠️  
![image](https://user-images.githubusercontent.com/50926159/116775795-4b7f8800-aa65-11eb-9382-d71e30fc2a26.png)